provider/openstack/provider.go

```diff
@@ -1147,23 +1147,31 @@ func (e *Environ) StartInstance(ctx context.ProviderCallContext, args environs.S
 			if err != nil {
 				break
 			}
+			if server == nil {
+				continue
+			}
 			var serverDetail *nova.ServerDetail
 			serverDetail, err = waitForActiveServerDetails(client, server.Id, 5*time.Minute)
-			if err != nil {
+			if err != nil || serverDetail == nil {
 				server = nil
 				break
 			} else if serverDetail.Status == nova.StatusActive {
 				break
 			} else if serverDetail.Status == nova.StatusError {
 				// Perhaps there is an error case where a retry in the same AZ
 				// is a good idea.
-				logger.Infof("Instance %q in ERROR state with fault %q", server.Id, serverDetail.Fault.Message)
-				logger.Infof("Deleting instance %q in ERROR state", server.Id)
+				faultMsg := " unable to determine fault details"
+				if serverDetail.Fault != nil {
+					faultMsg = fmt.Sprintf(" with fault %q", serverDetail.Fault.Message)
+				} else {
+					logger.Debugf("getting activer server details from nova failed without fault details")
```

Comment:
nit s/activer/active

Reply:
Oh good pick! I have just moved the code around without noticing it has a typo :) will fix.